### PR TITLE
do not show no data message outside indicator

### PR DESCRIPTION
**Pull request by waracci**
## Description

We should not indicate "no data available" outside of an indicator 

BUG:

Screenshot from 2021-06-27 13-40-42

FIX:

Screenshot from 2021-06-27 13-39-53

## Related Issue

https://trello.com/c/XRoppZAn/805-why-are-we-saying-no-data-available-outside-of-an-indicator-timebox-3-hours

## How to test it locally

Check for a category that has some indicators with data and others without.
This PR ensures that the "no data available"  message is not placed outside of the indicators without data

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally
- [ ] 👩‍🎨 does the design matches the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out
- [ ] commit messages are meaningful

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

**Comment by waracci** [comment]
[why are we saying "no data available" outside of an indicator [Timebox 3 hours]](https://trello.com/c/XRoppZAn/805-why-are-we-saying-no-data-available-outside-of-an-indicator-timebox-3-hours)

File: src/js/profile/subcategory.js [hunk 1]
```diff
@@ -16,120 +14,123 @@ const descriptionClass = '.sub-category-header__description';
...
-            'profile.chart.saveAsPng', 'profile.chart.valueTypeChanged',
-            'profile.chart.download_csv', 'profile.chart.download_excel', 'profile.chart.download_json', 'profile.chart.download_kml',
-            'point_tray.subindicator_filter.filter'
-        ]);
+    return block;
+  }
 
-        return block;
+  displayNoDataMsg(wrapper, msg, sc) {
+    if (!this._indicatorHasNoData[sc]) {
```
Comment (michaelglenister) [inline src/js/profile/subcategory.js:74]: So this method will only display a 'no data' response if this._indicatorHasNoData is correctly set? At the moment I don't see this set anywhere

**Review by michaelglenister — COMMENTED** [review]
Overall some refactoring was good, one issue is unclear regarding this._indicatorHasNoData, please clear that up, 
I see that adding spaces to the whole file creates some noise. Something I need to take note of myself

File: src/js/profile/subcategory.js [hunk 2]
```diff
@@ -16,120 +14,123 @@ const descriptionClass = '.sub-category-header__description';
...
+      let msgBox = document.createElement('p');
+      $(msgBox).addClass('msg-box').text(msg);
+      $(wrapper).append(msgBox);
+      this._indicatorHasNoData[sc] = true
```
Comment (waracci) [inline src/js/profile/subcategory.js:78]: @michaelglenister This is where we set the value for `this._indicatorHasNoData`

This ensures that the indicator logs a single message only.

**Comment by waracci** [comment]
> Overall some refactoring was good, one issue is unclear regarding this._indicatorHasNoData, please clear that up,
> I see that adding spaces to the whole file creates some noise. Something I need to take note of myself

Done. I added a pr comment indicating where it is being set.

I do not think we have set up any linting, any sort of formatting sort of gets through the pr review and merging pipeline. Linting and lining checks would be a great improvement.

**Comment by emre2038** [comment]
this PR doesn't fix the problem. created https://github.com/OpenUpSA/wazimap-ng-ui/pull/488 to replace this PR

**Pull request closed without merging**